### ISSUE TITLE
[Compiler Bug]: eslint-plugin-react-compiler versioning strategy breaks semver

### ISSUE THREAD

**Issue by lukpsaxo** [facebook/react#31173]
### What kind of issue is this?

- [ ] React Compiler core (the JS output is incorrect, or your app works incorrectly after optimization)
- [ ] babel-plugin-react-compiler (build issue installing or using the Babel plugin)
- [X] eslint-plugin-react-compiler (build issue installing or using the eslint plugin)
- [ ] react-compiler-healthcheck (build issue installing or using the healthcheck script)

### Repro steps

semver sorts versions alpha-numerically but the version of each release is 0.0.0-experimental-HASH

So basically it is sorted by the hash... which makes renovate want to downgrade the compiler and ignore new versions.

Can you change it to

0.0.1-experimental-date-HASH

with the 0.0.1 from now on in order to break out from the current bad versioning strategy?

**Comment by poteto** [comment]
@lukpsaxo you should be pinning babel-plugin-react-compiler to a specific version rather than a `^` which should solve your issue. The fix would not be to modify our versioning strategy, which we've chosen for consistency with regular React experimental releases

**Comment by lukeapage** [comment]
@poteto thanks for replying.

we do pin. The problem is renovate creates PRs to update all the time and it picks the last release according to semver - which because of the broken format is not the latest release.
And i don’t believe it’s a renovate “bug” - it picks the highest alpha numeric value..

**Comment by poteto** [comment]
@lukeapage looking at the renovate docs, you should be able to configure it such that it either ignores the compiler packages altogether, or pass a regex to only allow a minimum version of >0.0.0?

**Comment by poteto** [comment]
I deleted those extra random tags which were accidentally pushed a while back, so I think this should fix your issue going forward since presumably now any dependency management bot will pick up the correct latest version.

**Comment by poteto** [comment]
Could you try using this [`extractversion`](https://docs.renovatebot.com/configuration-options/#) option? The reason I don't want to change our experimental package names is for consistency, we have the exact same versioning pattern for react. Either way, I think the best course here is to try out the various options provided in renovate to see if you can fix it for your repo instead of changing this upstream which has implications on packages that depend on React experimental releases.

I will close this for now.

**Comment by lukpsaxo** [comment]
Thanks for the suggestion, I tried extractversion, it didn't work but I did come up with a config that does work - hopefully this comment helps someone.

```
        {
            matchPackagePatterns: ['eslint-plugin-react-compiler'],
            followTag: 'latest',
        },
```

The `followTag` makes it ignore the versions published and just pickup the tag value as the latest version.
